### PR TITLE
Remove implicit waits and clean up page objects

### PR DESCRIPTION
**Pull request by bobsilverberg**
Also introduced a `timeout` fixture to centralize the setting of the default timeout

Adhoc at http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/fxapom.adhoc/5/

@mozilla/web-qa-sorcerers r?

### CODE REVIEW

File: fxapom/pages/page.py [hunk 1]
```diff
@@ -2,86 +2,19 @@
...
-from selenium.common.exceptions import ElementNotVisibleException
-
-
-TIMEOUT = 10
 
 
 class Page(object):
 
-    def __init__(self, base_url, selenium):
+    def __init__(self, base_url, selenium, timeout):
```
Comment (davehunt) [inline fxapom/pages/page.py]: We should make this optional and set a default value, perhaps 10?

File: tests/test_login.py [hunk 2]
```diff
@@ -12,8 +13,9 @@ class TestLogin(object):
...
 
-    def test_user_can_sign_in(self, base_url, selenium, dev_account, click_login):
-        fxa = WebDriverFxA(base_url, selenium)
+    def test_user_can_sign_in(self, base_url, selenium, dev_account, click_login, timeout):
+        fxa = WebDriverFxA(base_url, selenium, timeout)
         fxa.sign_in(dev_account.email, dev_account.password)
-        WebDriverWait(selenium, 20).until(
-            lambda s: s.find_element(*self._fxa_logged_in_indicator_locator).is_displayed())
+        # We sometimes need to wait longer than the standard 10 seconds
+        Wait(selenium, 20).until(
```
Comment (davehunt) [inline tests/test_login.py]: I would just set the default to 20 seconds. It saves having two values, and it's not really going to cost us much.

**Comment by davehunt** [comment]
Great stuff, feels so good to be getting rid of these implicit waits. Be sure to release a new version after this merges. r+ with comments addressed! :+1:

**Comment by bobsilverberg** [comment]
Thanks for the review @davehunt. I made a few changes related to a default timeout. Could you please take another look?

**Comment by bobsilverberg** [comment]
I think for now I prefer the importing of `TIMEOUT` in `page.py` because it allows me to define that number in one place only, so I'm going to merge this as it currently stands. Thanks for the review @davehunt.

**Pull request merged**